### ISSUE TITLE
Possible Expansion of SpawnTypes

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/SpongeAPI#1157]
After handling SpongePowered/SpongeCommon#502, I've noticed a lot of variances in the type of spawning that can take place. I'd like to find out if plugin developers are looking for more detail or more finer grained understanding of *why* an `Entity` spawned. With the phases, it's pretty easy to pinpoint why an `Entity` spawned, depending on the circumstance.

**Comment by kashike** [comment]
If there are bleeding changes for `SpawnTypes`, I'd recommend renaming `CUSTOM` to `UNKNOWN`.
> Usually, Sponge can decipher the spawn type but in some cases, the type is just unknown.

**Comment by Wolfizen** [comment]
`SpawnTypes.ENTITY_SPLIT` or similar could be added for when entities split or otherwise create entities directly (no spawn eggs, etc). The prime example being slimes, of course. Currently, I see no reasonable `SpawnType` to represent a slime splitting.

**Comment by ryantheleach** [comment]
When you ask for whether they want finer grained, do you mean then you have already in the current cause system?

**Comment by gabizou** [comment]
@ryantheleach yes. There are a few cases that simply stating a `SpawnTypes.DROP_ITEM` is quite ambiguous still.

**Comment by ryantheleach** [comment]
I'm thinking this should change to a 'mega-tracking/request' thread and solve this piecemeal over time.
